src/System.Net.Sockets/src/System/Net/Sockets/Socket.Tasks.cs

```diff
@@ -854,10 +795,294 @@ internal AsyncTaskMethodBuilder<TResult> GetCompletionResponsibility(out bool re
         /// <summary>A SocketAsyncEventArgs with an associated async method builder.</summary>
         private sealed class Int32TaskSocketAsyncEventArgs : TaskSocketAsyncEventArgs<int>
         {
-            /// <summary>A cached, successfully completed task.</summary>
-            internal Task<int> _successfullyCompletedTask;
             /// <summary>Whether exceptions that emerge should be wrapped in IOExceptions.</summary>
             internal bool _wrapExceptionsInIOExceptions;
         }
+
+        /// <summary>A SocketAsyncEventArgs that can be awaited to get the result of an operation.</summary>
+        internal sealed class AwaitableSocketAsyncEventArgs : SocketAsyncEventArgs, IValueTaskSource, IValueTaskSource<int>
```

Comment:
Is this class a copy/paste from src/System.Net.Sockets/src/System/Net/Sockets/NetworkStream.cs or are there changes here we should review?

Reply:
The one from NetworkStream was moved here, then augmented with the interfaces.  I removed it from NetworkStream along with its CopyToAsync override, as with these changes and with the base Stream.CopyToAsync being updated to use the new overloads, the overridden CopyToAsync wasn't adding enough additional value to be worth the extra code/complexity.